include/libvfio-user.h

```diff
@@ -501,7 +501,13 @@ typedef struct {
     /*
      * Migration state transition callback.
      *
-     * The callback should return -1 on error, setting errno.
+     * The callback should return 0 on success and -errno on error.
+     *
+     * When operating in non-blocking mode (LIBVFIO_USER_FLAG_ATTACH_NB was
+     * passed to vfu_create_ctx) and -EBUSY is returned, transitioning to the
+     * new state becomes asynchronous: libvfio-user does not send a response to
+     * the client and does not process any new messages.  Transitioning to the
+     * new device state is completed by calling vfu_migr_done.
```

Comment:
should this explain what could happen if you call e.g. vfu_dma_read() ?
Or one day, with tcp-ip transport, vfu_irq_trigger() ?

Reply:
Yes we should explicitly state that vfu_dma_read/write and vfu_irq_trigger are not allowed.